persist-cli/src/main/java/io/ballerina/persist/utils/SqlScriptGenerationUtils.java

```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.persist.utils;
+
+import io.ballerina.persist.PersistToolsConstants;
+import io.ballerina.persist.objects.BalException;
+import io.ballerina.persist.objects.Entity;
+import io.ballerina.persist.objects.FieldMetaData;
+import io.ballerina.persist.objects.Relation;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Sql script generator.
+ *
+ * @since 0.1.0
+ */
+public class SqlScriptGenerationUtils {
+
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String TAB = "\t";
+    private static final String EMPTY = "";
+    private static final String PRIMARY_KEY_START_SCRIPT = NEW_LINE + TAB + "PRIMARY KEY(";
+    private static final String UNIQUE_KEY_START_SCRIPT = NEW_LINE + TAB + "UNIQUE KEY(";
+    private static final String UNIQUE = " UNIQUE";
+    private static final String ON_DELETE_SYNTAX = " ON DELETE";
+    private static final String ON_UPDATE_SYNTAX = " ON UPDATE";
+    private static final String RESTRICT = "persist:RESTRICT";
+    private static final String CASCADE = "persist:CASCADE";
+    private static final String SET_NULL = "persist:SET_NULL";
+    private static final String NO_ACTION = "persist:NO_ACTION";
+    private static final String RESTRICT_SYNTAX = " RESTRICT";
+    private static final String CASCADE_SYNTAX = " CASCADE";
+    private static final String NO_ACTION_SYNTAX = " NO ACTION";
+    private static final String SET_NULL_SYNTAX = " SET NULL";
+    private static final String SET_DEFAULT_SYNTAX = " SET DEFAULT";
+
+    private SqlScriptGenerationUtils(){}
+
+    public static void generateSqlScript(ArrayList<Entity> entityArray, Path absoluteSourcePath) throws BalException {
+        HashMap<String, List<String>> referenceTables = new HashMap<>();
+        List<String> tableNamesInScript = new ArrayList<>();
+        generateSqlScript(entityArray, referenceTables, absoluteSourcePath, tableNamesInScript);
```

Comment:
Why is 2 params passed into this? Is it being updated?

Reply:
Refactored this impl too [27f56f0](https://github.com/ballerina-platform/persist-tools/pull/60/commits/27f56f00a2995030037b8512dfb7e9848701706a). Can you review latest changes?